src/main/java/org/opentripplanner/middleware/bugsnag/BugsnagDispatcher.java

```diff
@@ -0,0 +1,161 @@
+import org.opentripplanner.middleware.bugsnag.response.Project;
+import org.opentripplanner.middleware.models.BugsnagEvent;
+import org.opentripplanner.middleware.models.BugsnagEventRequest;
+import org.opentripplanner.middleware.utils.FileUtils;
+import org.opentripplanner.middleware.utils.HttpUtils;
+import org.opentripplanner.middleware.utils.JsonUtils;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+import static org.opentripplanner.middleware.spark.Main.getConfigPropertyAsInt;
+import static org.opentripplanner.middleware.spark.Main.getConfigPropertyAsText;
+
+/**
+ * Class to provide Bugsnag information bu constructing required API calls to Bugsnag
+ */
+public class BugsnagDispatcher {
+
+    private static final String BUGSNAG_API_URL = "https://api.bugsnag.com";
+    private static final String BUGSNAG_API_KEY = getConfigPropertyAsText("BUGSNAG_API_KEY");
+    private static final String BUGSNAG_ORGANIZATION = getConfigPropertyAsText("BUGSNAG_ORGANIZATION");
+    private static final int BUGSNAG_REPORTING_WINDOW_IN_DAYS
+        = getConfigPropertyAsInt("BUGSNAG_REPORTING_WINDOW_IN_DAYS", 14);
+
+    private static Organization ORGANIZATION = null;
+    public static HashMap<String, Project> PROJECTS = new HashMap<>();
+
+    private static HashMap<String, String> BUGSNAG_HEADERS = new HashMap<>();
+
+    private static String EVENT_DATA_REQUEST_FILTER;
+
+    private static int CONNECTION_TIMEOUT_IN_SECONDS = 5;
+
+
+    static {
+        BUGSNAG_HEADERS.put("Authorization", "token " + BUGSNAG_API_KEY);
+        BUGSNAG_HEADERS.put("Accept", "application/json; version=2");
+        BUGSNAG_HEADERS.put("Content-Type", "application/json");
+
+        getOrganization();
+        getProjects();
+
+        EVENT_DATA_REQUEST_FILTER
+            = FileUtils.getFileContents("src/main/resources/org/opentripplanner/middleware/eventDataRequestFilter.json");
+
+        // replace placeholder with reporting window number of days
+        EVENT_DATA_REQUEST_FILTER
+            = EVENT_DATA_REQUEST_FILTER.replace("$BUGSNAG_REPORTING_WINDOW_IN_DAYS$",
+            String.valueOf(BUGSNAG_REPORTING_WINDOW_IN_DAYS));
+    }
+
+    /**
+     * Make the initial event data request to Bugsnag. This triggers an asynchronous job to prepare the data for one
+     * single download. The returned event data request can be used to check the status of the request
+     */
+    public static BugsnagEventRequest makeEventDataRequest() {
+
+        if (ORGANIZATION == null) {
+            return null;
+        }
+
+        URI uri = HttpUtils.buildUri(BUGSNAG_API_URL,
+            "/organizations/" + ORGANIZATION.id + "/event_data_requests",
+            null);
+
+        String response = HttpUtils.httpRequest(uri,
+            CONNECTION_TIMEOUT_IN_SECONDS,
+            HttpUtils.REQUEST_METHOD.POST,
+            BUGSNAG_HEADERS, EVENT_DATA_REQUEST_FILTER);
```

Comment:
To improve readability, let's have consistent line breaks after each comma. Also, the first param could be after a line break.

Reply:
Updated.